src/Google.Storage.V1/CopyObjectOptions.cs

```diff
@@ -0,0 +1,162 @@
+using static Google.Apis.Storage.v1.ObjectsResource.RewriteRequest;
+
+namespace Google.Storage.V1
+{
+    /// <summary>
+    /// Options for <c>CopyObject</c> operations.
+    /// </summary>
+    public class CopyObjectOptions
+    {
+        // TODO: maxBytesRewrittenPerCall? Specific to rewrite...
+
+        /// <summary>
+        /// The generation of the object resource to copy. When not
+        /// set, the latest generation will be copied.
+        /// </summary>
+        public long? SourceGeneration { get; set; }
+
+        /// <summary>
+        /// The projection to retrieve.
+        /// </summary>
+        public Projection? Projection { get; set; }
+
+        /// <summary>
+        /// A predefined set of ACLs for the new object.
+        /// </summary>
+        public PredefinedAcl? DestinationPredefinedAcl { get; set; }
+
+        /// <summary>
+        /// Precondition for copying: the object is only copied if the existing destination object's
+        /// generation matches the given value.
+        /// </summary>
+        public long? IfGenerationMatch { get; set; }
+
+        /// <summary>
+        /// Precondition for copying: the object is only copied if the existing destination object's
+        /// generation does not match the given value.
+        /// </summary>
+        public long? IfGenerationNotMatch { get; set; }
+
+        /// <summary>
+        /// Precondition for copying: the object is only copied if the existing destination object's
+        /// meta-generation matches the given value.
+        /// </summary>
+        public long? IfMetagenerationMatch { get; set; }
+
+        /// <summary>
+        /// Precondition for copying: the object is only copied if the existing destination object's
+        /// meta-generation does not match the given value.
+        /// </summary>
+        public long? IfMetagenerationNotMatch { get; set; }
+
+        /// <summary>
+        /// Precondition for copying: the object is only copied if the source object's
+        /// generation matches the given value.
+        /// </summary>
+        public long? IfSourceGenerationMatch { get; set; }
+
+        /// <summary>
+        /// Precondition for copying: the object is only copied if the source object's
+        /// generation does not match the given value.
+        /// </summary>
+        public long? IfSourceGenerationNotMatch { get; set; }
+
+        /// <summary>
+        /// Precondition for copying: the object is only copied if the source object's
+        /// meta-generation matches the given value.
+        /// </summary>
+        public long? IfSourceMetagenerationMatch { get; set; }
+
+        /// <summary>
+        /// Precondition for copying: the object is only copied if the source object's
+        /// meta-generation does not match the given value.
+        /// </summary>
+        public long? IfSourceMetagenerationNotMatch { get; set; }
+
+        internal void ModifyRequest(RewriteRequest request)
```

Comment:
Given that it *will* always be "options" (consistently, across all methods) I'm happier with the hard-coding in this case.